gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyLiveContentsPerExecutor.java

```diff
@@ -0,0 +1,258 @@
+    // Use a Spliterator to limit the processed commits to the "live" commits - i.e. stop traversing
+    // the expired commits once an entry is seen for each live content key.
+    new Spliterators.AbstractSpliterator<LogResponse.LogEntry>(src.estimateSize(), 0) {
+      private boolean more = true;
+
+      @Override
+      public boolean tryAdvance(Consumer<? super LogResponse.LogEntry> action) {
+        if (!more) {
+          return false;
+        }
+        more =
+            src.tryAdvance(
+                logEntry -> {
+                  // traverse until all the live commit heads are found for each live keys
+                  // at the time of cutoff time.
+                  if (foundAllLiveCommitHeadsBeforeCutoffTime.isFalse()) {
+                    action.accept(logEntry);
+                  }
+                  lastVisitedHash.set(logEntry.getCommitMeta().getHash());
+                });
+        more = more && foundAllLiveCommitHeadsBeforeCutoffTime.isFalse();
+        return more;
+      }
+    }.forEachRemaining(commitHandler);
+
+    return lastVisitedHash.get();
+  }
+
+  private LiveContentsResult computeLiveContents(
+      Instant cutOffTimestamp, String reference, Instant droppedRefTime, long bloomFilterSize) {
+    NessieApiV1 api = GCUtil.getApi(gcParams.getNessieClientConfigs());
+    TaskContext.get()
+        .addTaskCompletionListener(
+            context -> {
+              LOGGER.info("Closing the nessie api for compute live contents task");
+              api.close();
+            });
+    LOGGER.debug("Computing live contents for {}", reference);
+    Reference ref = GCUtil.deserializeReference(reference);
+    boolean isRefDroppedAfterCutoffTimeStamp =
+        droppedRefTime == null || droppedRefTime.compareTo(cutOffTimestamp) >= 0;
+    if (!isRefDroppedAfterCutoffTimeStamp) {
+      // reference is dropped before cutoff time.
+      // All the contents for all the keys are expired.
+      return ImmutableLiveContentsResult.builder()
+          .lastLiveCommitHash(ref.getHash())
+          .referenceName(ref.getName())
+          .hashOnReference(ref.getHash())
+          .bloomFilter(null)
+          .build();
+    }
+    Predicate<CommitMeta> liveCommitPredicate =
+        commitMeta ->
+            // If the commit time is newer than (think: greater than or equal to) cutoff-time,
+            // then commit is live.
+            commitMeta.getCommitTime().compareTo(cutOffTimestamp) >= 0;
+
+    ImmutableGCStateParamsPerTask gcStateParamsPerTask =
+        ImmutableGCStateParamsPerTask.builder()
+            .api(api)
+            .reference(ref)
+            .liveCommitPredicate(liveCommitPredicate)
+            .bloomFilterSize(bloomFilterSize)
+            .build();
+
+    return walkLiveCommitsInReference(gcStateParamsPerTask);
+  }
+
+  private LiveContentsResult walkLiveCommitsInReference(GCStateParamsPerTask gcStateParamsPerTask) {
```

Comment:
`GCStateParamsPerTask` is still needed for `handleLiveCommit()` as it has too many arguments. 

So, I feel no need to inline this function  (`walkLiveCommitsInReference`) because the caller will become too many lines of code and we don't get any extra benefits from inlining.

Reply:
Keeping `GCStateParamsPerTask` for the sake of 4 or so arguments feels odd.